src/send.rs

```diff
@@ -29,7 +29,7 @@ impl<const N: usize> SendBuffer<N> {
 
     /// Returns the number of bytes available in the buffer.
     pub fn available(&self) -> usize {
-        N - self.pending.iter().fold(0, |acc, x| acc + x.len()) + self.offset
+        N.saturating_sub(self.pending.iter().fold(0, |acc, x| acc + x.len()) + self.offset)
```

Comment:
This should already be fine, the issue with recv.rs is that it minus's the offset.

if you want you can try
`` N - self.pending.iter().fold(0, |acc, x| acc + x.len()) + self.offset``

Reply:
I see I messed up those brackets

Reply:
I don't think we should use saturing_sub/add since if there is a bug in the code this will hide it. If there is no order bug then this should never panic